### PR TITLE
disable the 'connect to a database' button if discovery + detection feature flag is on

### PR DESCRIPTION
**Pull request by adamsachs**
Partially closes [PROD-2138](https://ethyca.atlassian.net/browse/PROD-2138?focusedCommentId=41424)

Related to https://github.com/ethyca/fidesplus/pull/1455

### Description Of Changes

Disables the 'connect to a database' button in the UI if the discovery + detection feature flag is on.

This reduces confusion for end users while we still in the process of phasing out the 'legacy' scanning and classification workflows: if a user is using detection + discovery, then they should not be using the old "connect to a database" functionality.

### Code Changes

* [x] disable the 'connect to a database' button in the UI if the discovery + detection feature flag is on

### Steps to Confirm

* [x] tested both positive and negative cases locally, see screenshot below for when the feature flag is enabled (and button is disabled):

image

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] if there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated
* If there are any database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!

[PROD-2138]: https://ethyca.atlassian.net/browse/PROD-2138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/admin-ui/src/features/dataset/DatabaseConnectForm.tsx [hunk 1]
```diff
@@ -245,7 +245,11 @@ const DatabaseConnectForm = () => {
                 colorScheme="primary"
                 type="submit"
                 isLoading={isSubmitting || isLoading}
-                isDisabled={isSubmitting || isLoading}
+                isDisabled={
+                  isSubmitting ||
+                  isLoading ||
+                  features.flags.dataDiscoveryAndDetection
+                }
```
Comment (NevilleS) [inline clients/admin-ui/src/features/dataset/DatabaseConnectForm.tsx:252]: This looks technically correct, but I think it'd be less confusing to go further up in the workflow and completely hide the "Connect to a database" option when this flag is enabled.

i.e this screen:
image

**Review by NevilleS — CHANGES_REQUESTED** [review]

**Comment by NevilleS** [comment]
Also, re: this comment:
> not sure if there's a way to disable a feature flag for particular cypress tests, or if i should just remove these tests since we want to deprecate this button altogether soon...

To be very thorough in my answer, since you're a thorough guy...
1. It's definitely *possible* to disable a flag for a single test. You could navigate to `/management/about` and disable the flag in the test itself and navigate back... or it'd also be possible to trigger the Redux action directly 😄 
2. It's very reasonable to also disable these the legacy tests with `it.skip` and `describe.skip` though. You're right, the go-forward is the new monitoring UI, so regressions in the legacy fashion isn't our primary concern. You have my blessing 👍

File: clients/admin-ui/src/features/dataset/DatabaseConnectForm.tsx (hunk 1 above)
Comment (adamsachs) [reply clients/admin-ui/src/features/dataset/DatabaseConnectForm.tsx]: adjusted this in https://github.com/ethyca/fides/pull/4972/commits/d74c03f9bb597d4d2c0fca29981f57340b00a730

**Comment by adamsachs** [comment]
thanks for your comments @NevilleS! i made adjustments per your suggestions and i think this is good to go now 👍

**Review by NevilleS — APPROVED** [review]
LGTM. I pushed a minor nit to just add a tooltip as a hint (mostly for ourselves!) since it's non-obvious why that button is disabled:
image

**Comment by adamsachs** [comment]
> LGTM. I pushed a minor nit to just add a tooltip as a hint (mostly for ourselves!) since it's non-obvious why that button is disabled: image


that's great, thanks for that polish! i also was thinking something like that would be nice, just wasn't confident enough in my FE abilities to venture there 😆

**Pull request merged**